tokio-trace-fmt/src/span.rs

```diff
@@ -186,8 +190,12 @@ impl<'a> Context<'a> {
             .ok()?
     }
 
-    pub(crate) fn new(store: &'a Store) -> Self {
-        Self { store }
+    pub(crate) fn new(store: &'a Store, new_visitor: &'a N) -> Self {
+        Self { store, new_visitor }
+    }
+
+    pub(crate) fn new_visitor(&self, writer: &'a mut fmt::Write, is_empty: bool) -> N::Visitor {
```

Comment:
This should actually be public --- the expectation is that it will be used by the functions provided to `on_event`:
```suggestion
    pub fn new_visitor(&self, writer: &'a mut fmt::Write, is_empty: bool) -> N::Visitor {
```

Reply:
Also, if the bound on the impl block is changed to a HRTB, I think the `'a` in the writer here can be removed:
```suggestion
    pub(crate) fn new_visitor(&self, writer: &mut fmt::Write, is_empty: bool) -> N::Visitor {
```